datafusion/sqllogictest/test_files/errors.slt

```diff
@@ -168,8 +168,9 @@ CREATE TABLE tab0(col0 INTEGER, col1 INTEGER, col2 INTEGER);
 statement ok
 INSERT INTO tab0 VALUES(83,0,38);
 
-query error DataFusion error: Arrow error: Divide by zero error
+query I
```

Comment:
This is an interesting change -- we short circuited the evaluation and now the error doesn't happen

THis happens in other areas and so I think this change is consistent with other parts of DataFusion as well